src/SPU.cpp

```diff
@@ -993,15 +1003,34 @@ void SPU::Mix(u32 spucycles)
     }
 
     Platform::Mutex_Lock(AudioLock);
-    while (OutputSamplePos < 16)
-    {
-        u8 fb = OutputSamplePos & 0xF;
-        u8 fa = 0x10 - fb;
-        s16 leftfinal = ((OutputLastSamples[0] * fa) + (output[0] * fb)) >> 4;
-        s16 rightfinal = ((OutputLastSamples[1] * fa) + (output[1] * fb)) >> 4;
+    if (output[0] != OutputLastSamples[0])
+        blip_add_delta(BlipLeft, BlipTimer, (int) output[0] - OutputLastSamples[0]);
+    if (output[1] != OutputLastSamples[1])
+        blip_add_delta(BlipRight, BlipTimer, (int) output[1] - OutputLastSamples[1]);
+    BlipTimer += MixInterval >> 1;
```

Comment:
You should be using spucycles here, MixInterval is the "new" interval for the next event, spucycles is the "current" interval for this event.